src/init.cpp

```diff
@@ -368,6 +368,7 @@ std::string HelpMessage(HelpMessageMode mode)
             "Warning: Reverting this setting requires re-downloading the entire blockchain. "
             "(default: 0 = disable pruning blocks, >%u = target size in MiB to use for block files)"), MIN_DISK_SPACE_FOR_BLOCK_FILES / 1024 / 1024));
     strUsage += HelpMessageOpt("-reindex", _("Rebuild block chain index from current blk000??.dat files on startup"));
+    strUsage += HelpMessageOpt("-spentindex", strprintf(_("Maintain a full spent index, used to query the spending txid and input index for an outpoint (default: %u)"), false)); // fSpentIndex (insight_explorer)
```

Comment:
Its useful to have the purpose of the index documented somewhere especially if we decide to go with just one flag `-insightexplorer` which will be simpler for end-users, rather than having to add multiple flags for all the required indices.

Reply:
@bitcartel, I hope this is what you had in mind: https://github.com/zcash/zcash/pull/3837/commits/f9df1a98775dc6d65c10ac3c40274fbd01f7227d (for #3837) and in the current PR: https://github.com/zcash/zcash/pull/3849/commits/6f6c56daf6d6599f901749e30d39934c8a454305#diff-e8db9b851adc2422aadfffca88f14c91R148

But did you mean document somewhere other than in the source code?